scripts/zmtrigger.pl.in

```diff
@@ -293,21 +293,39 @@ while( 1 ) {
     }
   }
 
-# If necessary reload monitors
-  if ( $needsReload || ((time() - $monitor_reload_time) > MONITOR_RELOAD_INTERVAL )) {
+  # Reload all monitors from the dB every MONITOR_RELOAD_INTERVAL
+  if ( (time() - $monitor_reload_time) > MONITOR_RELOAD_INTERVAL ) {
     foreach my $monitor ( values(%monitors) ) {
-# Free up any used memory handle
-      zmMemInvalidate( $monitor );
+      zmMemInvalidate( $monitor ); # Free up any used memory handle
     }
     loadMonitors();
-    $needsReload = 0;
+    @needsReload = (); # We just reloaded all monitors so no need reload a specific monitor
+  # If we have NOT just reloaded all monitors, reload a specific monitor if its shared mem changed
+  } elsif ( @needsReload ) { 
+    foreach my $monitor ( @needsReload ) {
+      loadMonitor($monitor);
+    }
+    @needsReload = ();
   }
+
   # zmDbConnect will ping and reconnect if neccessary
   $dbh = zmDbConnect();
 } # end while ( 1 )
 Info( "Trigger daemon exiting\n" );
 exit;
 
+sub loadMonitor {
+  my $monitor = shift;
+
+  Debug( "Loading monitor $monitor\n" );
```

Comment:
Don't need the carriage return. Also monitor is a hash right? Maybe print out the I'd and name

Reply:
No, $monitor is a scalar. I put the carriage return in there for better readability. I want separate my initial definitions from the rest of the code.

Reply:
No I'm referring to the \n in the Debug.

Reply:
That line was copy/paste from the loadMonitors sub

Reply:
Wow... all the Info, Debug, etc statements in zmtrigger have \n